### PR TITLE
tests: fix i18n-test bugs in Node 13

### PR DESCRIPTION
**Pull request by brendankenny**
helps check off some of #10582

Two test issues here:
1. Node 13 has a `Intl.NumberFormat` bug where a `NaN` that was stored in a property on an object will be formatted as `-NaN`, e.g. `({a: NaN}).a.toLocaleString() === '-NaN'`. This also happens in V8 7.9/8.0 (~Chrome 79 and 80) but works correctly before and after these versions.

   Formatting `NaN`s isn't exactly important to our report, we added the test to exercise the corner cases and make sure they wouldn't cause a crash, so this PR adds a narrow workaround to the test that can be removed later. Try not to put `NaN`s in the report or they'll be negative for Node 13 users :)

2. The thousands separator for Spanish was removed starting in Node 13, causing the swap-locale formatted-number test to fail. Looking for NaN bugs I ran across [this crbug](https://bugs.chromium.org/p/chromium/issues/detail?id=1019268#c5), where it turns out that "Linguist experts in CLDR require two digits before the group separator for Spanish" (so a number has to be ≥ 10000 before getting a separator) which was fixed in V8 by an ICU update at some point.

   Unfortunately we don't have any 5+ digit formatted number in `sample_v2.json` to get the separator back, so this just switches the test to swap the locale to `de`, which *does* have 1 as its [`minimumGroupingDigits`](https://unicode.org/reports/tr35/tr35-numbers.html#Examples_of_minimumGroupingDigits).

### CODE REVIEW

**Comment by brendankenny** [comment]
R: @paulirish since he was also looking at these failures

File: lighthouse-core/test/lib/i18n/i18n-test.js [hunk 1]
```diff
@@ -298,7 +298,15 @@ describe('i18n', () => {
...
-      expect(helloNaNStr).toBeDisplayString('Hello NaN World');
+      // TODO: workaround can be removed after Node 13 is retired.
+      // expect(helloNaNStr).toBeDisplayString('Hello NaN World');
+
+      // Node 13/V8 7.9 and 8.0 have a bug where `({a: NaN}).a.toLocaleString() === "-NaN"`. It
+      // works correctly in Node 12 and 14, so work around it since NaN isn't essential for
+      // user-facing strings and it will eventually correct itself.
+      const formattedNaNStr = i18n.getFormatted(helloNaNStr, 'en-US');
+      expect(formattedNaNStr === 'Hello NaN World' || formattedNaNStr === 'Hello -NaN World')
+        .toBe(true);
```
Comment (brendankenny) [inline lighthouse-core/test/lib/i18n/i18n-test.js:309]: originally I had this as a conditional
```js
if (({a: NaN}).a.toLocaleString() === '-NaN') {
  expect(helloNaNStr).toBeDisplayString('Hello -NaN World');
} else {
  expect(helloNaNStr).toBeDisplayString('Hello NaN World');
}
```
but the conditional was always going to the `else` clause and then failing on the `Hello NaN World` assertion, so I gave up on figuring that out :) I assume some codegen thing.

File: lighthouse-core/test/lib/i18n/i18n-test.js [hunk 2]
```diff
@@ -298,7 +298,15 @@ describe('i18n', () => {
       expect(helloInfinityStr).toBeDisplayString('Hello ∞ World');
 
       const helloNaNStr = str_(UIStrings.helloBytesWorld, {in: NaN});
-      expect(helloNaNStr).toBeDisplayString('Hello NaN World');
+      // TODO: workaround can be removed after Node 13 is retired.
```
Comment (paulirish) [inline lighthouse-core/test/lib/i18n/i18n-test.js:301]:
```suggestion
      // TODO(COMPAT): workaround can be removed after Node 13 is retired.
```

File: lighthouse-core/test/lib/i18n/swap-locale-test.js [hunk 3]
```diff
@@ -11,27 +11,27 @@ const lhr = require('../../results/sample_v2.json');
 
 /* eslint-env jest */
 describe('swap-locale', () => {
-  it('can change golden LHR english strings into spanish', () => {
+  it('can change golden LHR english strings into german', () => {
```
Comment (patrickhulce) [inline lighthouse-core/test/lib/i18n/swap-locale-test.js:14]: 🎉 

image
